### PR TITLE
Handle RootCAs certificate

### PR DESCRIPTION
**Pull request by Juliens**
Add the possibility to configure self-signed certificate for tls communication with the backends

### CODE REVIEW

File: integration/https_test.go [hunk 1]
```diff
@@ -271,6 +271,46 @@ func (s *HTTPSSuite) TestWithClientCertificateAuthenticationMultipeCAsMultipleFi
...
+	defer cmd.Process.Kill()
+
+	// wait for Traefik
+	err = try.GetRequest("http://127.0.0.1:8080/api/providers", 1000*time.Millisecond, try.BodyContains("https://127.0.0.1:9040"))
+	c.Assert(err, checker.IsNil)
+
+	err = try.GetRequest("http://127.0.0.1:8081/ping", 1000*time.Millisecond, try.StatusCodeIs(http.StatusOK))
+
+	c.Assert(err, checker.IsNil)
+
```
Comment (ldez) [inline integration/https_test.go]: remove this blank line

File: integration/https_test.go [hunk 2]
```diff
@@ -271,6 +271,46 @@ func (s *HTTPSSuite) TestWithClientCertificateAuthenticationMultipeCAsMultipleFi
...
```
Comment (ldez) [inline integration/https_test.go]: remove this blank line

File: server/configuration.go [hunk 3]
```diff
@@ -113,6 +114,64 @@ func (dep *DefaultEntryPoints) Type() string {
...
+	}
+	return content, nil
+}
+
+// String is the method to format the flag's value, part of the flag.Value interface.
+// The String method's output will be used in diagnostics.
+func (dep *RootCAs) String() string {
+	sliceOfString := make([]string, len([]FileOrContent(*dep)))
+	for key, value := range *dep {
+		sliceOfString[key] = string(value)
```
Comment (ldez) [inline server/configuration.go]: `FileOrContent` need to implement `Stringer` instead of `string(foc)`

File: server/configuration.go [hunk 4]
```diff
@@ -113,6 +114,64 @@ func (dep *DefaultEntryPoints) Type() string {
...
+// RootCAs hold the CA we want to have in root
+type RootCAs []FileOrContent
+
+// FileOrContent hold a file path or content
+type FileOrContent string
+
+func (f FileOrContent) Read() ([]byte, error) {
+	var content []byte
+	if _, err := os.Stat(string(f)); err == nil {
+		content, err = ioutil.ReadFile(string(f))
```
Comment (ldez) [inline server/configuration.go]: `FileOrContent` need to implement `Stringer` instead of `string(foc)`

File: integration/https_test.go [hunk 5]
```diff
@@ -271,6 +271,46 @@ func (s *HTTPSSuite) TestWithClientCertificateAuthenticationMultipeCAsMultipleFi
...
+	err := cmd.Start()
+	c.Assert(err, checker.IsNil)
+	defer cmd.Process.Kill()
+
+	// wait for Traefik
+	err = try.GetRequest("http://127.0.0.1:8080/api/providers", 1000*time.Millisecond, try.BodyContains("https://127.0.0.1:9040"))
+	c.Assert(err, checker.IsNil)
+
+	err = try.GetRequest("http://127.0.0.1:8081/ping", 1000*time.Millisecond, try.StatusCodeIs(http.StatusOK))
+
```
Comment (ldez) [inline integration/https_test.go]: remove this blank line

File: integration/https_test.go [hunk 6]
```diff
@@ -271,6 +271,46 @@ func (s *HTTPSSuite) TestWithClientCertificateAuthenticationMultipeCAsMultipleFi
...
```
Comment (ldez) [inline integration/https_test.go]: remove this blank line

File: cmd/traefik/traefik.go [hunk 7]
```diff
@@ -180,6 +183,22 @@ func run(traefikConfiguration *server.TraefikConfiguration) {
 		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 	}
 
+	if len(globalConfiguration.RootCAs) > 0 {
+		roots := x509.NewCertPool()
+		for _, cert := range globalConfiguration.RootCAs {
+			certContent, err := cert.Read()
+			if err != nil {
+				log.Error("Error while read RootCAs", err)
```
Comment (emilevauge) [inline cmd/traefik/traefik.go]: `continue` ?

**Pull request merged**